### PR TITLE
DRILL-7523: Update Calcite to 1.31.0

### PR DESCRIPTION
**Pull request by vvysotskyi**
# [DRILL-7523](https://issues.apache.org/jira/browse/DRILL-7523): Update Calcite to 1.31.0

## Description
Changes required for updating Calcite to 1.31.0. For now, the master version built by JitPack is used. After Calcite 1.31.0 is released, this PR will be updated to use that version.

No Drill-specific commits in Calcite are required after this change, so the https://github.com/vvysotskyi/drill-calcite repository will be archived after merging this PR.

## Documentation
NA

## Testing
All unit tests pass.

### CODE REVIEW

**Comment by cgivre** [comment]
@vvysotskyi This is really amazing work!  Are you keeping this in draft until Calcite releases version 1.31, or is there additional work you are working on?

**Comment by vvysotskyi** [comment]
@cgivre, yes, I'm keeping it in draft until the Calcite 1.31 release, no additional work is required for this PR.

**Comment by vvysotskyi** [comment]
Now it has the latest official released Calcite (and Avativa) version, so it is ready for review.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlSelectBuilder.java [hunk 1]
```diff
@@ -90,8 +91,14 @@ public SqlSelectBuilder fetch(SqlNode fetch) {
     return this;
   }
 
+  public SqlSelectBuilder hints(SqlNodeList hints) {
```
Comment (cgivre) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlSelectBuilder.java:94]: @vvysotskyi Will Drill do anything with the SQL hints that Calcite now supports?

Comment (vvysotskyi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlSelectBuilder.java:94]: In the future, I think we will use this functionality to provide more flexibility for configuring the planning process.

**Review by cgivre — APPROVED** [review]
I don't know that I am qualified to review the actual code changes here, but they look good to me.  However, I tested this manually on my local machine and noticed significant speed improvements.  That combined with the unit tests passing leads me to say: LGTM +1.

Thanks @vvysotskyi for completing this really complex PR and getting us back in line with main Calcite.

**Pull request merged**